packages/sns/src/sns.wrapper.ts

```diff
@@ -378,6 +381,13 @@ export class SnsWrapper {
   ): Promise<BuyerState | undefined> =>
     this.swap.getUserCommitment(this.mergeParams(params));
 
+  getOpenTicket = (params: Omit<QueryParams, "certified">): Promise<Result_1> =>
```

Comment:
The pattern we normally use is return the data and throw an error if get an error message.

Reply:
For example the `manage_neuron` related methods in the sns governance. You did it in the `splitNeuron` I believe as well

Reply:
Btw, you might want to expose the return from `notifyParticipation` as well.

Just so that you already have it. At the moment it returns `void`.

Reply:
- Custom error types for ticket api (not sure about having an error type per error code, imo too much code, but not a strong opinion):  https://github.com/dfinity/ic-js/pull/280/files#diff-2e190f3116fa6ade09f9fac59a3d8ffd3ccd9c05484ff3e829d6253bde2d1acbR7
- expose of the RefreshBuyerTokensResponse: https://github.com/dfinity/ic-js/pull/280/files#diff-a5bf8c90276cb97f9642f5635c72af8c7206837d9193cac86f25f1b168500f08R378